### PR TITLE
Adding legba (network bruteforcer)

### PR DESCRIPTION
**Pull request by cHJlaXpoZXI**
# Description

I added the tool legba (https://github.com/evilsocket/legba) a login bruteforcer / password spraying written in Rust

# Related issues

No related issue.

# Point of attention

I hope that everything is good :)

### CODE REVIEW

**Review by ShutdownRepo — APPROVED** [review]
Impressive, the PR is near perfection on the first try, thank you!
Two minor changes and we should be good to go
EIDT : I messed up and **approved** instead of **requesting changes**, this is a change request review

File: sources/install/package_network.sh [hunk 1]
```diff
@@ -248,6 +248,22 @@ function install_rustscan() {
     add-to-list "rustscan,https://github.com/RustScan/RustScan,The Modern Port Scanner"
 }
 
+function install_legba() {
+    # CODE-CHECK-WHITELIST=add-aliases
+    colorecho "Installing legba"
+    git -C /opt/tools/ clone --depth 1 https://github.com/evilsocket/legba
+    cd /opt/tools/legba || exit
+    # Sourcing rustup shell setup, so that rust binaries are found when installing cme
+    source "$HOME/.cargo/env"
```
[the remark this answers is not in the document]
Comment (cHJlaXpoZXI) [reply sources/install/package_network.sh]: I have deleted lines.

File: sources/assets/shells/history.d/legba
[the remark this answers is not in the document]
Comment (cHJlaXpoZXI) [reply sources/assets/shells/history.d/legba:1]: I have modified as expected :)

**Comment by ShutdownRepo** [comment]
@cHJlaXpoZXI we now have this issue in the build, do you know how we could fix that so that it doesn't block the upcoming release?
```
2023-12-19T22:18:14.9103120Z #23 214.2   cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
2023-12-19T22:18:14.9104500Z #23 214.2   cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
2023-12-19T22:18:14.9105600Z #23 214.2   error: SMB Client library not found! Probably libsmbclient is not installed.
2023-12-19T22:18:14.9106510Z #23 214.2 
2023-12-19T22:18:14.9106960Z #23 214.2   --- stderr
2023-12-19T22:18:14.9108290Z #23 214.2   thread 'main' panicked at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pavao-0.2.3/build.rs:28:13:
2023-12-19T22:18:14.9110590Z #23 214.2   `PKG_CONFIG_ALLOW_SYSTEM_CFLAGS="1" PKG_CONFIG_ALLOW_SYSTEM_LIBS="1" "pkg-config" "--libs" "--cflags" "smbclient"` did not exit successfully: exit status: 1
2023-12-19T22:18:14.9112480Z #23 214.2   error: could not find system library 'smbclient' required by the 'pavao' crate
2023-12-19T22:18:14.9113340Z #23 214.2 
2023-12-19T22:18:14.9113720Z #23 214.2   --- stderr
2023-12-19T22:18:14.9114500Z #23 214.2   Package smbclient was not found in the pkg-config search path.
2023-12-19T22:18:14.9115540Z #23 214.2   Perhaps you should add the directory containing `smbclient.pc'
2023-12-19T22:18:14.9116480Z #23 214.2   to the PKG_CONFIG_PATH environment variable
2023-12-19T22:18:14.9117510Z #23 214.2   Package 'smbclient', required by 'virtual:world', not found
2023-12-19T22:18:14.9118280Z #23 214.2 
2023-12-19T22:18:14.9119420Z #23 214.2   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
2023-12-19T22:18:14.9120620Z #23 214.2 warning: build failed, waiting for other jobs to finish...
2023-12-19T22:18:27.7901350Z #23 ERROR: process "/bin/sh -c ./entrypoint.sh package_network" did not complete successfully: exit code: 101
```

**Comment by cHJlaXpoZXI** [comment]
Hi !

After some research on the project, we need to add this two packets : libsmbclient-dev libsmbclient.  They can be installed though apt package. It's because EvilSocket added the support for SMB protocol.

I have modified on my fork but the PR is closed so I cannot add it here. I can do another PR if you want.

**Comment by ShutdownRepo** [comment]
Will add it myself, thank you! 🙏 
I will come back here if something fails

**Pull request merged**